python/across_api/base/fov.py

```diff
@@ -0,0 +1,354 @@
+    distance = hpcoord.separation(skycoord).to(u.deg)
+
+    # Create the probability density distribution HEALPix map
+    prob = stats.norm(scale=error_radius).pdf(distance)
+
+    # Normalize it
+    prob /= np.sum(prob)
+
+    return prob
+
+
+class FOVBase:
+    visible_pixels: np.ndarray
+
+    def probability_in_fov(
+        self,
+        skycoord: Optional[SkyCoord] = None,
+        error_radius: Optional[u.Quantity[u.deg]] = None,
+        healpix_loc: Optional[FITS_rec] = None,
+    ) -> float:
+        """
+        For a given sky position and error radius, calculate the probability of
+        the sky position being inside the field of view (FOV).
+
+        Parameters
+        ----------
+        skycoord
+                SkyCoord object representing the position of the celestial object.
+        error_radius
+                The error radius for the sky position. If not given, the `skycoord`
+                will be treated as a point source.
+        healpix_loc
+                HEALPix map of the localization.
+        time
+                The time of the observation.
+        ephem
+                Ephemeris object.
+
+        """
+        # For a point source
+        if skycoord is not None and (error_radius is None or error_radius == 0.0):
+            in_fov = self.in_fov(skycoord)
+            return 1.0 if in_fov else 0.0
+        # For a circular error region
+        if skycoord is not None and error_radius is not None:
+            return self.in_fov_circular_error(
+                skycoord=skycoord, error_radius=error_radius
+            )
+        # For a HEALPix map
+        elif healpix_loc is not None:
+            return self.in_fov_healpix_map(healpix_loc=healpix_loc)
+
+        # We should never get here
+        raise AssertionError("No valid arguments provided")
+
+    def in_fov(
+        self,
+        skycoord: SkyCoord,
+    ) -> Union[bool, np.ndarray]:
+        """
+        Is a coordinate or set of coordinates `skycoord` inside the FOV and not
+        Earth occulted.
+
+        Note that this method only checks if the given coordinate is Earth
+        occulted, so defines a simple 'all-sky' FOV with no other constraints.
+        For more complex FOVs, this method should be overridden with one that
+        also checks if coordinate is inside the bounds of an instrument's FOV
+        for a given spacecraft attitude.
+
+        Parameters
+        ----------
+        skycoord
+                SkyCoord object representing the celestial object.
+        time
+                Time object representing the time of the observation.
+        ephem
+                Ephemeris object
+
+        Returns
+        -------
+        bool
+                True or False
+        """
+
+        # Check if skycoord pixels in list of visible pixels
+        skycoord_pix = hp.ang2pix(
+            hp.order2nside(HEALPIX_MAP_EVAL_ORDER),
+            skycoord.ra.value,
+            skycoord.dec.value,
```

Comment:
Don't assume that the units are degrees.

```suggestion
            skycoord.ra.deg,
            skycoord.dec.deg,
```